hi-ml/src/health_ml/utils/data_augmentations.py

```diff
@@ -171,17 +171,34 @@ def __init__(self, kernel_size: int, p: float = 0.5, min: float = 0.1, max: floa
         self.kernel_size = kernel_size
         self.p = p
 
-    def __call__(self, sample: torch.Tensor) -> torch.Tensor:
+    @staticmethod
+    def apply_gaussian_blur(sample: torch.Tensor, kernel_size: int, p: float, min: float, max: float) -> torch.Tensor:
         prob = np.random.random_sample()
-
-        if prob < self.p:
-            sigma = (self.max - self.min) * np.random.random_sample() + self.min
-            sample = np.array(sample.squeeze())  # type: ignore
-            sample = cv2.GaussianBlur(sample, (self.kernel_size, self.kernel_size), sigma)
-            sample = torch.Tensor(sample).unsqueeze(0)
-
+        if prob < p:
+            sigma = (max - min) * np.random.random_sample() + min
+            sample = sample.permute([0, 2, 3, 1]).squeeze().numpy()   # only 3 channels, color channel last
+            sample = cv2.GaussianBlur(sample, (kernel_size, kernel_size), sigma)
+            sample = torch.Tensor(sample).unsqueeze(0).permute(0, 3, 1, 2)
         return sample
 
+    def __call__(self, img: torch.Tensor) -> torch.Tensor:
+        original_shape = img.shape
+        if len(original_shape) == 3:
+            img = img.unsqueeze(0)  # add batch dimension if missing
+        # if the input is a bag of images, gaussian blur needs to run on each image separately
+        if img.shape[0] > 1:
+            for i in range(img.shape[0]):
+                img_tile = img[i]
+                img[i] = self.apply_gaussian_blur(sample=img_tile.unsqueeze(0), kernel_size=self.kernel_size, p=self.p,
```

Comment:
Maybe I don't get something important here, feel free to ping me for a chat, but: I think this is sequence of operations is very wasteful. Apply_gaussian_blur can deal with tensors that have a batch dimension. Why are we going over the image batch here, effectively removing the batch dimension, then adding it back on via unsqueeze?

Reply:
The opencv Gaussian blur works with single images, similar to our stain normalization transform. we are applying similar checks for stain normalization.

Reply:
I agree that the squeeze and unsqueeze operations could be simplified.

Reply:
I'll open another PR for the changes, since it affects both stain norm and Gaussian blur transforms